app/models/ticket_purchase.rb

```diff
@@ -58,4 +63,24 @@ def self.update_quantity(conference, quantity, ticket, user)
     purchase.quantity = quantity if quantity > 0
     purchase
   end
+
+  def pay(payment)
+    self.update_attributes(paid: true, payment_id: payment.id)
+  end
+
+  protected
+
+  def create_physical_tickets
```

Comment:
I agree with Ana, call the **pay** method in the purchase_ticket after creating the **TicketPurchase** object if it is free. That way you will have all the logic in one place, the **pay** method.

But rather than just doing **purchas.pay** you will need to do **purchase.pay(nil)** in the case of free tickets.